src/runtime/storageNG/storage-proxy.ts

```diff
@@ -21,19 +29,47 @@ export class StorageProxy<T extends CRDTTypeRecord> {
   private crdt: CRDTModel<T>;
   private id: number;
   private store: ActiveStore<T>;
+  private listenerAttached = false;
+  private keepSynced = false;
+  private synchronized = SyncState.none;
+  private readonly scheduler: StorageProxyScheduler<T>;
 
-  constructor(crdt: CRDTModel<T>, store: ActiveStore<T>) {
+  constructor(
+      crdt: CRDTModel<T>,
+      store: ActiveStore<T>,
+      scheduler: StorageProxyScheduler<T>) {
```

Comment:
Looks like this is always an instance of the real scheduler; does it need to be passed in (i.e. could just be constructed inline)?

Reply:
The reason I got it as a parameter is so I can keep a reference in the test (to wait all work is done before verifying stuff). Alternative would to expose a method to get a reference to the scheduler. Let me know if that would be better and I can change it.